internal/k8s/config.go

```diff
@@ -0,0 +1,33 @@
+package k8s
+
+import (
+	"github.com/hashicorp/go-hclog"
+
+	"github.com/hashicorp/consul-api-gateway/internal/core"
+	"github.com/hashicorp/consul-api-gateway/internal/k8s/gatewayclient"
+	"github.com/hashicorp/consul-api-gateway/internal/k8s/reconciler"
+	"github.com/hashicorp/consul-api-gateway/internal/store"
+)
+
+func StoreConfig(adapter core.SyncAdapter, client gatewayclient.Client, logger hclog.Logger, config Config) store.Config {
```

Comment:
This just exists to simplify the construction of the dependencies (deployer, status updater, etc.)